README.rst

```diff
@@ -70,49 +117,61 @@ websocket implementation. For this example, plug in your Sanic server.
 
     app.run(host="0.0.0.0", port=8000)
 
-And then, plug into a subscribable schema:
 
-.. code:: python
+Django v2+
+~~~~~~~~~~
 
-    import asyncio
-    import graphene
 
+Django Channels 2
+~~~~~~~~~~~~~~~~~
 
-    class Query(graphene.ObjectType):
-        base = graphene.String()
+Set up with Django Channels just takes three steps:
 
+1. Install the apps
+2. Set up your schema
+3. Configure the channels router application
 
-    class Subscription(graphene.ObjectType):
-        count_seconds = graphene.Float(up_to=graphene.Int())
+First ``pip install channels`` and it to your ``INSTALLED_APPS``. If you
+want graphiQL, install the ``graphql_ws.django`` app before
+``graphene_django`` to serve a graphiQL template that will work with
+websockets:
 
-        async def resolve_count_seconds(root, info, up_to):
-            for i in range(up_to):
-                yield i
-                await asyncio.sleep(1.)
-            yield up_to
+.. code:: python
 
+    INSTALLED_APPS = [
+        "channels",
+        "graphql_ws.django",
+        "graphene_django",
+        # ...
+    ]
 
-    schema = graphene.Schema(query=Query, subscription=Subscription)
+Point to your schema in Django settings:
 
-You can see a full example here:
-https://github.com/graphql-python/graphql-ws/tree/master/examples/aiohttp
+.. code:: python
 
-Gevent
-~~~~~~
+    GRAPHENE = {
+        'SCHEMA': 'yourproject.schema.schema'
```

Comment:
this tripped me up: you should mention in the doc that it is required for the channel work to function where it is technically not needed for graphene to work.

I had to add to my existing project.

Reply:
This is under the django channels configuration section, so it would imply that it's the setup required for channels, wouldn't it? Maybe I'm not understanding - please elaborate

Reply:
This is a setup I found on the Django-graphene website: https://docs.graphene-python.org/projects/django/en/latest/tutorial-plain/#update-settings

It is "new"ish and, so far, works without it unless you activate the subscriptions (not sure why).

I point it out for anyone like me who has a working project setup before the django-graphene update that works as-is: subscriptions will not work and the error is not that clear.

It is a suggestion because I chased my tail for ~20mins looking at an opaque error ;-)